hircluster.h

```diff
@@ -117,6 +117,7 @@ typedef struct redisClusterContext {
     struct timeval *connect_timeout;            /* TCP connect timeout */
     struct timeval *command_timeout;            /* Receive and send timeout */
     int max_retry_count;                        /* Allowed retry attempts */
+    char *username;                             /* Authenticate using user */
     char password[CONFIG_AUTHPASS_MAX_LEN + 1]; /* Include a null terminator */
```

Comment:
Here they differ too. It looks a bit messy. What's the point of having password fixed length when it's optional? Maybe we shall change that too to an allocated string? Or make username fixed length?

Reply:
Yes, I think changing password to a pointer is better.
It seems that there is no given limit on the length of username in Redis, so guessing it would probably not be right.
It also seems that Redis own CONFIG_AUTHPASS_MAX_LEN is not used to limit the password length anymore on master..so I guess it could be handling passwords longer than 512 these days (!)

It also makes it easier to debug using pointers since today the RedisClusterContext is larger than the default object sizelimit in gdb.